### ISSUE TITLE
Output of promtool unit tests is hard to read

### ISSUE THREAD

**Issue by lelutin** [prometheus/prometheus#15423]
### Proposal

Currently the text that's output when a test is failing is pretty hard for a human to parse. This is an example of a test failure I'm working on today that takes a good amount of effort to identify the problem -- only one number differs between the two in the annotations:

~~~
  FAILED:
    alertname: DonateHighFailureRate, time: 1h1m,
        exp:[
            0:
              Labels:{alertname="DonateHighFailureRate"}
              Annotations:{description="More than 1 failed transaction per minute for the last hour\nhas been detected on the production donate.torproject.org\nsite. Last increase per 10 minutes is 1.\n\nThis might be card testing and should be investigated\npromptly.\n", playbook="https://gitlab.torproject.org/tpo/tpa/team/-/wikis/service/donate#stripe-
card-testing", summary="Unusually high failure rate on donate.torproject.org"}
            ],
        got:[
            0:
              Labels:{alertname="DonateHighFailureRate", severity="warning"}
              Annotations:{description="More than 1 failed transaction per minute for the last hour\nhas been detected on the production donate.torproject.org\nsite. Last increase per 10 minutes is 120.\n\nThis might be card testing and should be investigated\npromptly.\n", playbook="https://gitlab.torproject.org/tpo/tpa/team/-/wikis/service/donate#strip
e-card-testing", summary="Unusually high failure rate on donate.torproject.org"}
            ]
~~~

if the output looked more like a diff, the difference could be made more obvious. the diff output could be made optional with a CLI flag to activated it, and if it looks useful enough it could become the default in a future major release only.

**Comment by lelutin** [comment]
hah ... and there's actually also a label difference that I had failed to spot

**Comment by crush-on-anechka** [comment]
> if the output looked more like a diff, the difference could be made more obvious

Hi! I decided to dig a little and discovered that the behavior you'd like to see can be achieved using this flag:

testRulesDiff := testRulesCmd.Flag("diff", "[Experimental] Print colored differential output between expected & received output.").Default("false").Bool()

**Comment by lelutin** [comment]
ah nice. indeed. thanks for searching @crush-on-anechka 

that was apparently merged in last january in #13342

apparently the reason I missed it out was that debian unstable still has promtool 2.45.6 which old by now. With the latest version in a container I can see and use the option.

I guess I'll turn around and poke debian into packaging a newer version.

cheers!
